### PR TITLE
added two way wave equation comparison

### PR DESCRIPTION
**Pull request by jbrzensk**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Example
- [ ] Documentation

## Description
Added a comparison for the two way wave equation for second order finite difference and mimetics. The comparsison includes metrics such as error, FLOPs, time needed, and cost of switching methods.

## Added/updated tests?
_We encourage you to test all code included with MOLE, including examples.

- [ ] Yes
- [x] No, and this is why: examples are tests
- [ ] I need help with writing tests

## Read Contributing Guide and Code of Conduct

- [x] I have read and followed the [Contributing Guide](https://github.com/csrc-sdsu/mole/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/csrc-sdsu/mole/blob/main/CODE_OF_CONDUCT.md)

## [optional] What gif best describes this PR or how it makes you feel?
I leave this here for Valeria!

### CODE REVIEW

**Comment by valeriabarra** [comment]
Hi @jbrzensk , thank you so much for this useful contribution! 
I ran in under `octave` (I run Ubuntu) - I am not sure whether it is an octave/MATLAB compatibility thing, but the graphs are illegible to me. The font of the labels and legends are tiny. The line sizes are also tiny. See an example below

image

File: examples/matlab_octave/compare_2way_wave_equation.m [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+% just the methods.
+%
+% The FTCS has a second order accurate time scheme, so it will never get >2
+% convergence. But, this show how easy it is to switch orders and
+% experiment, for very little cost.
+% 2025/08/18
+clc
+close all
+clear
+
```
[the remark this answers is not in the document]
Comment (jbrzensk) [reply examples/matlab_octave/compare_2way_wave_equation.m:24]: Updated in latest commit fd930fe

**Comment by jbrzensk** [comment]
Updates...

File: examples/matlab_octave/compare_2way_wave_equation.m [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    % Mimetic Difference, same as above, just staggered grid
+    U0_md = u(grid_md', 0);
+
+    analytic_md = u(grid_md', (t+1)*dt);
+
+    %% FTCS Scheme Matrix L
+    % For step 1
+    % Make a vector of ones, this will be the -1,+1 diagonal of matrix FTCS
+    B = ones(nx,1);
+    B = ( r2_fd ) * B;
```
Comment (aboada) [inline examples/matlab_octave/compare_2way_wave_equation.m:88]: For simplicity, I think you can say:
```
B = ( r2_fd ) * ones(nx,1);
```

**Review by aboada — COMMENTED** [review]
Thanks for the example. Added some minor comments on formatting and style. Even though it is a MATLAB code, I suggest looking into adding functions to extract functionality, which will improve the readability and maintainability of the code.

File: examples/matlab_octave/tutorials/two_way_wave_equation/comparison_two_way_wave_md_vs_fd.m [hunk 3]
```diff
@@ -0,0 +1,154 @@
+%% COMPARISON_TWO_WAY_WAVE_MD_VS_FD
```
[the remark this answers is not in the document]
Comment (jbrzensk) [reply examples/matlab_octave/tutorials/two_way_wave_equation/comparison_two_way_wave_md_vs_fd.m:1]: Added. We have been placing them at the bottom to avoid distracting users running the ```help commandXX```  in MATLAB.

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
+# Tutorial: Comparing Finite Difference and Mimetic Difference Operators
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:1]: Thank you for adding this, @jbrzensk ! I would remove the "Tutorial:" part in the title, as it sounds like a repetition in the TOC ("Tutoals/Tutorial: ...."). I would also give it a shorter name. Perhaps a quick
 ```suggestion
# Mimetic Difference Operators  Vs Finite Differences
```
?

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
+# Tutorial: Comparing Finite Difference and Mimetic Difference Operators  
+### Solving the 1D Two-Way Wave Equation
+
+This tutorial demonstrates how to solve the **1D two-way wave equation** using both **standard finite differences (FD)** and **mimetic finite differences (MD)**.  
+We will use three main functions:
+
+1. [`finite_diff_two_way_wave_eq`](#finite_diff_two_way_wave_eq)
+2. [`mimetic_diff_two_way_wave_eq`](#mimetic_diff_two_way_wave_eq)
+3. [`comparison_two_way_wave_md_vs_fd`](#comparison_two_way_wave_md_vs_fd)
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:9]: These links to headers do not seem to work. Check how to escape special characters (`_`)

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+The goal is to understand the **differences in accuracy, stability, and numerical properties** between traditional and mimetic operators. The two numerical functions have been created to be as similar as possible, so that only the changes necessary for mimetic difference operators are visible.
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:11]:
```suggestion
The goal is to understand the **differences in accuracy, stability, and numerical properties** between traditional finite and mimetic difference operators. The two numerical functions have been created to be as similar as possible, so that only the changes necessary for mimetic difference operators are visible.
```

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+We assume no boundary effects (large domain), allowing a clean comparison of the spatial discretization methods. Here is a little bit more information about the two seperate functions.
+
+---
+
+## 1. `finite_diff_two_way_wave_eq`
+
+**Purpose:** This function solves the the wave equation using **standard finite differences (FD)**.
+
+**Time-stepping scheme (CTCS):**
+If we discretize the equation with standard centered second order finite differences, Equation(1) turn into
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:36]:
```suggestion
If we discretize the equation with standard centered second order finite differences, Equation(1) turns into 
```

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+### Function Outputs
+| Variable | Description |
+|-----------|--------------|
+| `U2_fd` | Final solution at the last time step |
+| `error_fd` | Norm of error vs. analytic/reference solution |
+| `walltime_fd` | Wall-clock time (seconds) |
+| `flops_fd` | Estimated floating-point operation count |
+
+### Notes
+We are using an explicit two-step scheme (leapfrog). The CFL condition must be satisfied: $ c \Delta t^2 / \Delta x^2 \le 1 $. If you change the stepping, or increase the number of cells, be sure to change this value. No boundary conditions are applied, the domain is large enough to avoid reflection effects. This is to test just the spacial scheme without any boundary considerations.
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:82]:
```suggestion
We are using an explicit two-step scheme (leapfrog). The CFL condition must be satisfied: $ c \Delta t^2 / \Delta x^2 \le 1 $. If you change the stepping, or increase the number of cells, be sure to change this value. No boundary conditions are applied, the domain is large enough to avoid reflection effects. This is to test just the spatial scheme without any boundary considerations.
```

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+
+---
+
+## 2. `mimetic_diff_two_way_wave_eq`
+
+**Purpose:** Solve the wave equation using **mimetic difference operators (MD)**.
+
+The mimetic Laplacian operator `L` replaces the explicit finite-difference stencil, automatically enforcing discrete analogs of conservation and symmetry properties.
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:90]: I would remove "explicit" here, as it is often used in regards to time-stepping, but we are talking about the spatial discretization here.
```suggestion
The mimetic Laplacian operator `L` replaces the finite-difference stencil, automatically enforcing discrete analogs of conservation and symmetry properties.
```

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+\frac{U^{k-1}_i - 2U^{k}_i + U^{k+1}_i}{\Delta t^2} = c^2 L\,(U^k)
+$$
+
+Here, `L` is constructed from mimetic gradient (`G`) and divergence (`D`) operators such that:
+
+$$
+L = D \, G
+$$
+
+ensuring energy and flux consistency at the discrete level. The mimetic library Laplacian operator is just DG under the hood.
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:105]:
```suggestion
ensuring energy and flux consistency at the discrete level. The mimetic library Laplacian operator is just the composition $DG$ under the hood.
```

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+
+**Purpose:** Solve the wave equation using **mimetic difference operators (MD)**.
+
+The mimetic Laplacian operator `L` replaces the explicit finite-difference stencil, automatically enforcing discrete analogs of conservation and symmetry properties.
+
+**Time-stepping scheme:**
+The mimetic operator directly takes the place fo the second derivative (Laplacian), therefore we only need to discretize the time derivative. Doing so to Equation (1) yields:
+
+$$
+\frac{U^{k-1}_i - 2U^{k}_i + U^{k+1}_i}{\Delta t^2} = c^2 L\,(U^k)
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:96]: Since you gave the form of the matrix above for the Finite Difference operator, I would expect here to see the form of the corresponding MD operator for completeness sake.

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+```
+
+This staggered layout provides **better discrete analogs** of differential operators, leading to improved conservation and often reduced numerical dispersion.
+
+---
+
+## Convergence and Accuracy
+
+Both FD and MD methods use a **second-order accurate** CTCS scheme in time.  
+The mimetic method, however, allows **higher-order spatial accuracy** by increasing the mimetic operator order `k`.
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:185]: I would add:
```suggestion
The mimetic method, however, allows **higher-order spatial accuracy** by increasing the mimetic operator order `k` without having to reconstruct the matrix.
```
?
Of something of that sort. FD can be high-order, too. You would just have to redefine your matrix.

File: doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+|----------|------------------|--------------------|
+| Grid | Uniform | Staggered |
+| Operator | Explicit stencil | Discrete mimetic operator |
+| Conservation | Approximate | Exact (discrete) |
+| Flexibility | Fixed 2nd order | Adjustable order (k) |
+| Accuracy | 2nd order | ≥ 2nd order |
+| Ease of implementation | Simple | Slightly more complex |
+| Best use case | Quick prototyping | Physically consistent PDE solvers |
+
+---
```
Comment (valeriabarra) [inline doc/sphinx/source/examples/Tutorials/Two_Way_Wave_Equation/Two_Way_Wave_Equation.md:215]: The description is very nice but we're not using the `literalinclude` clause to actually show snippets of code. 
We should also investigate if an automatic execution and compilation of the tutorial is possible (showing any output, such as plots), like it is for Jupyter notebook for instance or dynamic languages such as python and Julia
